widgets/common/src/filters/RuleRow.tsx

```diff
@@ -22,9 +22,7 @@ const RuleRow: FunctionComponent<RuleRowProps> = ({ id, onRemove }) => {
             <FormField>
                 <RuleInput />
             </FormField>
-            <FormField>
-                <RuleRemoveButton onClick={() => onRemove(id)} />
-            </FormField>
+            <FormField>{allowRemove && <RuleRemoveButton onClick={onRemove} />}</FormField>
```

Comment:
Optional, if it is possible :smile: 

Does it matter that we render an empty `FormField`, as opposed to not rendering it at all?

```suggestion
            {allowRemove && <FormField><RuleRemoveButton onClick={onRemove} /></FormField>}
```

This will render 1 fewer element :smile:

Reply:
No, it's not good in this case (length of the row is changing then). I'll keep it as it is.

Reply:
Makes sense :slightly_smiling_face: